packages/react/src/primitives/Autocomplete/useAutocomplete.tsx

```diff
@@ -0,0 +1,253 @@
+import * as React from 'react';
+
+import {
+  ARROW_DOWN,
+  ARROW_UP,
+  ENTER_KEY,
+  ESCAPE_KEY,
+} from '../shared/constants';
+import { isFunction } from '../shared/utils';
+import { useStableId } from '../utils/useStableId';
+import type { Option, UseAutocompleteProps } from '../types';
+
+const DEFAULT_KEYS = new Set([ARROW_DOWN, ARROW_UP, ENTER_KEY, ESCAPE_KEY]);
+
+export const useAutocomplete = ({
```

Comment:
I do not expose this hook but it sounds like a good idea. Let me move the combobox props into the hook, so that it returns everything people need to build a custom autocomplete